coalib/settings/ConfigurationGathering.py

```diff
@@ -99,59 +98,55 @@ def save_sections(sections):
-def warn_nonexistent_targets(targets, sections, log_printer):
+def warn_nonexistent_targets(targets, sections):
     """
     Prints out a warning on the given log printer for all targets that are
     not existent within the given sections.
 
     :param targets:     The targets to check.
     :param sections:    The sections to search. (Dict.)
-    :param log_printer: The log printer to warn to.
     """
     for target in targets:
         if target not in sections:
-            log_printer.warn(
+            logging.warning(
                 "The requested section '{section}' is not existent. "
                 'Thus it cannot be executed.'.format(section=target))
 
     # Can't be summarized as python will evaluate conditions lazily, those
     # functions have intended side effects though.
-    files_config_absent = warn_config_absent(sections, 'files', log_printer)
-    bears_config_absent = warn_config_absent(sections, 'bears', log_printer)
+    files_config_absent = warn_config_absent(sections, 'files')
+    bears_config_absent = warn_config_absent(sections, 'bears')
     if files_config_absent or bears_config_absent:
         raise SystemExit(2)  # Invalid CLI options provided
 
 
-def warn_config_absent(sections, argument, log_printer):
+def warn_config_absent(sections, argument):
     """
     Checks if the given argument is present somewhere in the sections and emits
     a warning that code analysis can not be run without it.
 
     :param sections:    A dictionary of sections.
     :param argument:    The argument to check for, e.g. "files".
-    :param log_printer: A log printer to emit the warning to.
     """
     if all(argument not in section for section in sections.values()):
-        log_printer.warn('coala will not run any analysis. Did you forget '
-                         'to give the `--{}` argument?'.format(argument))
+        logging.warning('coala will not run any analysis. Did you forget '
+                        'to give the `--{}` argument?'.format(argument))
         return True
 
     return False
 
 
-def load_configuration(arg_list, log_printer, arg_parser=None, args=None):
+def load_configuration(arg_list, arg_parser=None, args=None):
     """
     Parses the CLI args and loads the config file accordingly, taking
     default_coafile and the users .coarc into account.
 
     :param arg_list:    The list of CLI arguments.
-    :param log_printer: The LogPrinter object for logging.
     :param arg_parser:  An ``argparse.ArgumentParser`` instance used for
                         parsing the CLI arguments.
     :param arg_list:    Alternative pre-parsed CLI arguments.
-    :return:            A tuple holding (log_printer: LogPrinter, sections:
-                        dict(str, Section), targets: list(str)). (Types
-                        indicated after colon.)
+    :return:            A tuple holding (sections: dict(str, Section),
+                        targets: list(str)). (Types indicated after colon.)
```

Comment:
I think it is related to this commit, since I'm removing one element (log_printer) from tuple.